CaloReco/src/CaloHitMakerFast_module.cc

```diff
@@ -0,0 +1,158 @@
+           caloDigisToken_    {consumes<CaloDigiCollection>(config().caloDigiCollection())},
+           digiSampling_      (config().digiSampling()),
+           deltaTPulses_      (config().deltaTPulses()),
+           pulseRatioMin_     (config().pulseRatioMin()),
+           pulseRatioMax_     (config().pulseRatioMax()),
+           caphriCrystalID_   (config().caphriCrystalID()),
+           diagLevel_         (config().diagLevel())
+        {
+            produces<CaloHitCollection>("calo");
+            produces<CaloHitCollection>("caphri");
+        }
+
+        void produce(art::Event& e) override;
+
+
+    private:
+        using pulseMapType = std::unordered_map<unsigned, std::vector<HitInfo>>;
+
+        void extractHits(const CaloDigiCollection& caloDigis, CaloHitCollection& caloHitsColl, CaloHitCollection& caphriHitsColl);
+        void addPulse(pulseMapType& pulseMap, unsigned crystalID, float time, float eDep);
+
+        art::ProductToken<CaloDigiCollection> caloDigisToken_;
+        double              digiSampling_;
+        double              deltaTPulses_;
+        double              pulseRatioMin_;
+        double              pulseRatioMax_;
+        std::vector<double> caphriCrystalID_;
+        int                 diagLevel_;
+    };
+
+
+
+    //--------------------------------------------------------------------------------------------------------------
+    void CaloHitMakerFast::produce(art::Event& event)
+    {
+        if (diagLevel_ > 0) std::cout<<"[FastRecoDigiFromDigi] begin"<<std::endl;
+
+        const auto& caloDigisHandle = event.getValidHandle(caloDigisToken_);
+        const auto& caloDigis(*caloDigisHandle);
+
+        auto caloHitsColl   = std::make_unique<CaloHitCollection>();
+        auto caphriHitsColl = std::make_unique<CaloHitCollection>();
+        extractHits(caloDigis,*caloHitsColl,*caphriHitsColl);
+
+        event.put(std::move(caloHitsColl),  "calo");
+        event.put(std::move(caphriHitsColl),"caphri");
+
+        if (diagLevel_ > 0) std::cout<<"[FastRecoDigiFromDigi] end"<<std::endl;
+        return;
+    }
+
+
+   //--------------------------------------------------------------------------------------------------------------
+   void CaloHitMakerFast::extractHits(const CaloDigiCollection& caloDigis, CaloHitCollection& caloHitsColl, CaloHitCollection& caphriHitsColl)
+   {
+       const Calorimeter& cal = *(GeomHandle<Calorimeter>());
+       ConditionsHandle<CalorimeterCalibrations> calorimeterCalibrations("ignored");
+
+       pulseMapType pulseMap;
+       for (const auto& caloDigi : caloDigis)
+       {
+           int crystalID   = cal.caloIDMapper().crystalIDFromSiPMID(caloDigi.SiPMID());
+
+           double baseline = (caloDigi.waveform().at(0)+caloDigi.waveform().at(1)+caloDigi.waveform().at(2)+caloDigi.waveform().at(3))/4.0; 
+           double eDep     = (caloDigi.waveform().at(caloDigi.peakpos())-baseline)*calorimeterCalibrations->ADC2MeV(caloDigi.SiPMID()); 
```

Comment:
I just realized that the ADC2MeV value in the caloCalibrations needs to be updated.